graph/pull.go

```diff
@@ -219,8 +219,11 @@ func (s *TagStore) pullRepository(r *registry.Session, out io.Writer, repoInfo *
 	}
 
 	logrus.Debugf("Retrieving the tag list")
-	tagsList, err := r.GetRemoteTags(repoData.Endpoints, repoInfo.RemoteName)
+	tagsList, err := r.GetRemoteTags(repoData.Endpoints, repoInfo.RemoteName, askedTag)
 	if err != nil {
+		if strings.Contains(err.Error(), "not found") && askedTag != "" {
```

Comment:
Could we add a type for this, while we're here? String matching will never work out the way you want.